plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -149,7 +150,14 @@ func findNodesThatFit(
 	nodeNameToInfo map[string]*schedulercache.NodeInfo,
 	nodes []*api.Node,
 	predicateFuncs map[string]algorithm.FitPredicate,
-	extenders []algorithm.SchedulerExtender) ([]*api.Node, FailedPredicateMap, error) {
+	extenders []algorithm.SchedulerExtender,
+	metaFactory interface{},
+) ([]*api.Node, FailedPredicateMap, error) {
+	// For the time being, the metadata factory is optional.  We may make it required at some point.
```

Comment:
certainly, but at the function level i dont think we are throwing an error  if its nil ... henc the comment.  Should i throw an error if its nil ?